src/pages/ThorChainLP/Pool/components/PoolChart.tsx

```diff
@@ -1,34 +1,130 @@
+
+    const tvlFiat = fromThorBaseUnit(poolTotalDepth).times(runePrice).toFixed()
+    return {
+      time: Number(interval.startTime),
+      value: Number(tvlFiat),
+    }
+  })
+
+// @ts-ignore we can't make this a partial record as we need to use this as a tuple to spread as useQuery params
+const INTERVAL_PARAMS_BY_INTERVAL: Record<Interval | 'all', [Interval, number]> = {
+  day: ['hour', 24],
+  week: ['hour', 24 * 7],
+  month: ['day', 7 * 4],
+  all: ['month', 24],
+}
+
+type PoolChartProps = {
+  thorchainNotationAssetId: string
+}
+export const PoolChart = ({ thorchainNotationAssetId }: PoolChartProps) => {
+  const [selectedInterval, setSelectedInterval] = useState<Interval | 'all'>('day')
+  const [selectedDataType, setSelectedDataType] = useState<'volume' | 'liquidity'>('volume')
+
+  const setSelectedVolumeDataType = useCallback(() => setSelectedDataType('volume'), [])
+  const setSelectedLiquidityDataType = useCallback(() => setSelectedDataType('liquidity'), [])
+
+  const runeMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, thorchainAssetId),
+  )
+
+  const { data: swapsData } = useQuery({
+    ...reactQueries.midgard.swapsData(
+      thorchainNotationAssetId,
+      ...INTERVAL_PARAMS_BY_INTERVAL[selectedInterval],
+    ),
+    select: data => swapHistoryToChartData(data, runeMarketData.price),
+  })
+
+  const { data: tvl } = useQuery({
+    ...reactQueries.midgard.tvl(...INTERVAL_PARAMS_BY_INTERVAL[selectedInterval]),
+    select: data => tvlToChartData(data, runeMarketData.price, thorchainNotationAssetId),
+  })
+
+  const data = useMemo(() => {
+    const maybeData = selectedDataType === 'volume' ? swapsData : tvl
+    return maybeData ?? []
+  }, [selectedDataType, swapsData, tvl])
 
-export const PoolChart = () => {
-  const translate = useTranslate()
   return (
-    <Stack>
-      <Flex justifyContent='space-between'>
-        <ButtonGroup size='sm' isDisabled>
-          <Button>Volume</Button>
-          <Button>Liquidity</Button>
+    <Stack spacing={4}>
+      <Flex justifyContent='space-between' alignItems='center' p={4}>
+        <ButtonGroup size='sm'>
+          <Button isActive={selectedDataType === 'volume'} onClick={setSelectedVolumeDataType}>
+            Volume
+          </Button>
+          <Button
+            isActive={selectedDataType === 'liquidity'}
+            onClick={setSelectedLiquidityDataType}
+          >
+            Liquidity
+          </Button>
         </ButtonGroup>
-        <ButtonGroup size='sm' isDisabled>
-          <Button>1M</Button>
-          <Button>1W</Button>
-          <Button>All</Button>
+        <ButtonGroup size='sm'>
+          {Object.keys(INTERVAL_PARAMS_BY_INTERVAL).map(interval => {
+            const label =
+              interval === 'all'
+                ? 'All'
+                : // For all others, we should only keep the first char, as uppercase
+                  interval.charAt(0).toUpperCase()
```

Comment:
We may be better off with a display name mapping for readability, but in an case `all` title case logic should also be applicable with `interval.charAt(0).toUpperCase()` since `'all'.charAt(0).toUpperCase() == 'All'`